Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -0,0 +1,1941 @@
+               'RiskScore', 'ExposureLevel']
+    username = demisto.args().get('username')
+    response = get_user_machines_request(username)
+    machines_list = get_machines_list(response)
+    hr = tableToMarkdown(f'Machines that are related to user {username}:', machines_list, headers=headers)
+    context_output = {
+        'Username': username,
+        'Machines': machines_list
+    }
+    ec = {
+        'MicrosoftATP.UserMachine(val.Username === obj.Username)': context_output
+    }
+    return hr, ec, response
+
+
+def add_remove_machine_tag_command():
+    """Adds or remove tag to a specific Machine.
+    Returns:
+        (str, dict, dict). Human readable, context, raw response
+    """
+    headers = ['ID', 'ComputerDNSName', 'OSPlatform', 'LastIpAddress', 'LastExternalIPAddress', 'HealthStatus',
+               'RiskScore', 'ExposureLevel']
+    machine_id = demisto.args().get('machine_id')
+    action = demisto.args().get('action')
+    tag = demisto.args().get('tag')
+    response = add_remove_machine_tag_request(machine_id, action, tag)
+    machine_data = get_machine_data(machine_id)
+    hr = tableToMarkdown(f'Succeed to {action} tag to {machine_id}:', machine_data, headers=headers)
+    ec = {
+        'MicrosoftATP.Machine(val.ID === obj.ID)': machine_data
+    }
+    return hr, ec, response
+
+
+def fetch_incidents():
+    last_run = demisto.getLastRun()
+    last_alert_fetched_time = get_last_alert_fetched_time(last_run)
+    existing_ids = last_run.get('existing_ids', [])
+    latest_creation_time = last_alert_fetched_time
+    filter_alerts_creation_time = create_filter_alerts_creation_time(last_alert_fetched_time)
+    alerts = list_alerts_request(filter_alerts_creation_time)['value']
+
+    incidents, new_ids, latest_creation_time = all_alerts_to_incidents(alerts, latest_creation_time, existing_ids)
+
+    demisto.setLastRun({
+        'last_alert_fetched_time': datetime.strftime(latest_creation_time, '%Y-%m-%dT%H:%M:%S.%f'),
+        'existing_ids': new_ids
+    })
+    demisto.incidents(incidents)
+
+
+def create_filter_alerts_creation_time(last_alert_fetched_time):
+    """Create filter with the last alert fetched time to send in the request
+    Args:
+        last_alert_fetched_time(date): Last date and time of alert that been fetched
+    Returns:
+        (str). The filter of alerts creation time that will be send in the  alerts list API request
+    """
+    filter_alerts_creation_time = f"alertCreationTime+gt+{last_alert_fetched_time.isoformat()}"
+
+    if not filter_alerts_creation_time.endswith('Z'):
+        filter_alerts_creation_time = filter_alerts_creation_time + "Z"
+
+    return filter_alerts_creation_time
+
+
+def all_alerts_to_incidents(alerts, latest_creation_time, existing_ids):
+    """Gets the alerts list and convert it to incidents
+    Args:
+        alerts(list): List of alerts filtered by the first_fetch_timestamp parameter
+        latest_creation_time(date):  La
```

Comment:
WHY?

Reply:
changed timestamp from ' %Y-%m-%dT%H:%M:%S.%f' to '%Y-%m-%dT%H:%M:%S' as we talked